core/shared/src/main/scala/scalaz/zio/ZIO.scala

```diff
@@ -1424,7 +1446,7 @@ trait ZIOFunctions extends Serializable {
    * Unlike `foreachAllPar`, this method will use at most `n` fibers.
    */
   final def collectAllParN[R >: LowerR, E <: UpperE, A](n: Long)(as: Iterable[ZIO[R, E, A]]): ZIO[R, E, List[A]] =
-    foreachParN[R, E, ZIO[R, E, A], A](n)(as)(identity(_))
+    foreachParN[R, E, ZIO[R, E, A], A](n)(as)(a => a)
```

Comment:
For performance reasons, we should consider doing this inside the `ZIO` object:

```scala
private val _IdentityFn: Any => Any = (a: Any) => a
private def identityFn[A]: A => A = _IdentityFn.asInstanceOf[A => A]
```

Then instead of `a => a`, you can use `ZIO.identityFn[A]`, which will save an allocation.

Reply:
then I think this should be propagated to the other usages as well. Right?

Reply:
Yes, any place inside `ZIO.scala` could use `identityFn` for increased performance. No need to go outside ZIO.scala, though.

Reply:
`identityFn`